src/library/constructions/no_confusion.cpp

```diff
@@ -101,6 +101,7 @@ static optional<environment> mk_no_confusion_type(environment const & env, name
                         expr rhs_type = lctx.get_type(rhs);
                         level l       = sort_level(type_checker(env, lctx).ensure_type(lhs_type));
                         expr h_type;
+                        // TODO: this should respect reducibility
```

Comment:
In lean 3, this was possible with `type_context_old ctx(env, transparency_mode::Reducible);`.

I think the issue here is that `mk_no_confusion_type` is using the kernel, when it should be using the elaborator; but fixing that is a large refactor.